.github/workflows/macos.yml

```diff
@@ -0,0 +1,81 @@
+name: MacOS tests
+
+on: [pull_request, push]
+
+permissions:
+  contents: read
+
+jobs:
+  macos:
+# Run a job for each of the specified os configs:
+    strategy:
+      matrix:
+        os:
+          - macos-13
+          - macos-12
+          - macos-11
+# eventually build for other alg sets, if CCI use is discontinued
+        params:
+          - oqsconfig: -DOQS_ALGS_ENABLED=STD
+    runs-on: ${{matrix.os}}
+    env:
+# Don't overwhelm github CI VMs:
+      MAKE_PARAMS: -j 4
+    steps:
+      - name: Checkout provider
+        uses: actions/checkout@v3
+      - name: Checkout openssl
+        uses: actions/checkout@v3
+        with:
+          set-safe-directory: true
+          repository: openssl/openssl
+          path: openssl
+      - name: checkout liboqs
+        uses: actions/checkout@v3
+        with:
+          set-safe-directory: true
+          repository: open-quantum-safe/liboqs
+          ref: main
+          path: liboqs
+      - name: Retrieve OpenSSL32 from cache
+        id: cache-openssl32
+        uses: actions/cache@v3
+        with:
+          path: .localopenssl32
+          key: ${{ runner.os }}-openssl32
+      - name: Build openssl3 if not cached
+        if: steps.cache-openssl32.outputs.cache-hit != 'true'
+        run: pwd && ./config --prefix=`pwd`/../.localopenssl32 && make $MAKE_PARAMS && make install_sw install_ssldirs
+        working-directory: openssl
+      - name: Save OpenSSL
+        id: cache-openssl-save
+        if: steps.cache-openssl32.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: |
+            .localopenssl32
+          key: ${{ runner.os }}-openssl32
+      - name: build liboqs
+        run: |
+           mkdir _build
+           cd _build
+           cmake -DOPENSSL_ROOT_DIR=../.localopenssl32 -DCMAKE_INSTALL_PREFIX=../../.localliboqs ${{ matrix.params.oqsconfig }}  ..
```

Comment:
I'm curious as to why you are using `make` for building liboqs rather than ninja or cmake --build, but no bother.

Reply:
No clue. "Code blindness"? Or: "If `make` works (only) for `openssl` also use it for `liboqs`"? Strange.